### PR TITLE
Atoms & Snippets - Remove business logic from templates

### PR DESCRIPTION
**Pull request by annebyrne**
The `notificationValue` function has been left in `storyQuestions` as its related to a test and involves string output. It will be removed after the test is over.

Tested on `CODE`

### CODE REVIEW

File: common/app/model/content/Atoms.scala [hunk 1]
```diff
@@ -199,7 +200,9 @@ final case class TimelineAtom(
 final case class TimelineItem(
   title: String,
   date: DateTime,
-  body: Option[String]
+  body: Option[String],
+  dateFormat: Option[String],
```
Comment (TBonnin) [inline common/app/model/content/Atoms.scala]: Is `dateFormat` ever being used?

**Review by TBonnin — APPROVED** [review]
I have one question but look good to me otherwise. Thank you for doing this @annebyrne

File: common/app/model/content/Atoms.scala [hunk 2]
```diff
@@ -553,12 +586,26 @@ object TimelineAtom {
...
+    item.toDate match {
+      case Some(date) => {
+        val beginRange = DateTimeFormat.forPattern(format).print(item.date)
+        val endRange = DateTimeFormat.forPattern(format).print(date)
+        s"$beginRange&ndash;$endRange"
+      }
+      case None => DateTimeFormat.forPattern(format).print(item.date)
+    }
+  }
+
```
Comment (TBonnin) [inline common/app/model/content/Atoms.scala]: Would writting unit test for this function help making sure the implemenation is right and more importantly isn't changed by accident?

File: common/app/views/fragments/atoms/snippets/timeline.scala.html [hunk 3]
```diff
@@ -28,9 +14,9 @@
       <div class="explainer-snippet__item">
         <time class="explainer-snippet__event-date" datetime="">
```
Comment (regiskuckaertz) [inline common/app/views/fragments/atoms/snippets/timeline.scala.html]: Semantically, each date should be its own `time` element. If you move the logic out from the template, let's go all the way have do:
```
<time class="explainer-snippet__event-date" datetime="@item.date.toMachineReadable">
  @item.data.toHumanReadable(item.dateFormat)
</time>@item.toDate.map { toDate =>–<time class="explainer-snippet__event-date" datetime="@toDate.toMachineReadable">@toDate.toHumanReadable(item.dateFormat)</item>
}

File: common/app/model/content/Atoms.scala [hunk 4]
```diff
@@ -515,7 +518,37 @@ object ReviewAtom {
...
 
 object StoryQuestionsAtom {
-  def make(atom: AtomApiAtom): StoryQuestionsAtom = StoryQuestionsAtom(atom.id, atom, atom.data.asInstanceOf[AtomData.Storyquestions].storyquestions)
+  def make(atom: AtomApiAtom): StoryQuestionsAtom = StoryQuestionsAtom(
+    atom.id,
+    atom,
+    atom.data.asInstanceOf[AtomData.Storyquestions].storyquestions
+  )
+
+  def isClosed(question: StoryQuestionsAtom): Boolean = {
```
Comment (regiskuckaertz) [inline common/app/model/content/Atoms.scala]: This should be used to filter out closed storyquestions and not worry about them later.

Comment (regiskuckaertz) [reply common/app/model/content/Atoms.scala]: You can update [`extract`](https://github.com/guardian/frontend/blob/0ea1950923d9765618118d57084ab48ce2b6469f/common/app/model/content/Atoms.scala#L209) to do that for you:
```
def extract[T](
  atoms: Option[Seq[AtomApiAtom]], 
  extractFn: AtomApiAtom => T,
  filter: AtomApiAtom => Boolean = const(false)): Seq[T] = {
    try {
      atoms.getOrElse(Nil).filterNot(filter).map(extractFn)
    } catch {
      case e: Exception =>
        logException(e)
        Nil
    }
}
```
and then
```scala
val storyquestions = extract(atoms.storyquestions, atom => { StoryQuestionsAtom.make(atom) }, StoryQuestionsAtom.isClosed)
```

**Comment by annebyrne** [comment]
Tested on `CODE` so going to merge :+1:

**Pull request merged**